src/polymer/polymer-element.ts

```diff
@@ -80,10 +81,61 @@ export interface Options {
   sourceRange: SourceRange|undefined;
 }
 
+export interface ScannedPolymerExtension extends ScannedElementBase {
```

Comment:
Arg, this is painful. Yeah, I can definitely see why we want mixins here.

Add a TODO to use native mixins ourselves here once we drop node 4?